lib/timeutils.c

```diff
@@ -502,35 +502,24 @@ int strtime_iso(const time_t *t, int flags, char *buf, size_t bufsz)
 	return -1;
 }
 
-/* relative time functions */
-int time_is_today(const time_t *t, struct timeval *now)
+int strtime_short(const time_t *t, struct timeval *now, int flags, char *buf, size_t bufsz)
 {
-	if (now->tv_sec == 0)
-		gettimeofday(now, NULL);
-	return *t / (3600 * 24) == now->tv_sec / (3600 * 24);
-}
+	struct tm tm, tmnow;
+	int rc = 0;
 
-int time_is_thisyear(const time_t *t, struct timeval *now)
-{
 	if (now->tv_sec == 0)
 		gettimeofday(now, NULL);
-	return *t / (3600 * 24 * 365) == now->tv_sec / (3600 * 24 * 365);
-}
-
-int strtime_short(const time_t *t, struct timeval *now, int flags, char *buf, size_t bufsz)
-{
-        struct tm tm;
-	int rc = 0;
 
-        localtime_r(t, &tm);
+	localtime_r(t, &tm);
+	localtime_r(&now->tv_sec, &tmnow);
 
-	if (time_is_today(t, now)) {
+	if (tm.tm_yday == tmnow.tm_yday && tm.tm_year == tmnow.tm_year) {
```

Comment:
Would be possible to keep is_today() and is_this_year() in the code to make it more readable? (We can use static inline functions or macros there.)

Reply:
A little sleep, and seeing that could be done better way: d393c00c6cd57ef7f122f4e1730b3c410b6084cb. Updated version calls time_is_thisyear() within time_is_today().